File: March 30, 2026.js
function getDueDate(dateStr) {

  let [year, month, day] = dateStr.split("-").map(Number);
  const date = new Date(year, month - 1, day);
  date.setDate(1);
  date.setMonth(date.getMonth() + 9);

  const lastDay = new Date(date.getFullYear(), date.getMonth() + 1, 0).getDate();
  date.setDate(Math.min(day, lastDay));
  year = date.getFullYear();
  month = String(date.getMonth() + 1).padStart(2, "0");
  day = String(date.getDate()).padStart(2, "0");


  return `${year}-${month}-${day}`;
}

File: March 30, 2026.py
from datetime import datetime
import calendar

def get_due_date(date_str):

    year, month, day = map(int, date_str.split("-"))
    date = datetime(year, month, day)

    year = date.year
    month = date.month
    
    month += 9
    year += (month - 1) // 12
    month = ((month - 1) % 12) + 1
    
    last_day = calendar.monthrange(year, month)[1]

    day = min(day, last_day)

    return f"{year:04d}-{month:02d}-{day:02d}"